src/sidebar/store/util.js

```diff
@@ -20,42 +20,62 @@ function actionTypes(updateFns) {
 function createReducer(...actionToUpdateFn) {
   // Combine the (action name => update function) maps together into a single
   // (action name => update functions) map.
+  //
+  // After namespace migration, remove the requirement for actionToUpdateFns
+  // to use arrays. Why? createReducer will be called once for each module rather
+  // than once for all modules.
   const actionToUpdateFns = {};
   actionToUpdateFn.forEach(map => {
     Object.keys(map).forEach(k => {
       actionToUpdateFns[k] = (actionToUpdateFns[k] || []).concat(map[k]);
     });
   });
 
-  return (state, action) => {
+  return (state = {}, action) => {
     const fns = actionToUpdateFns[action.type];
     if (!fns) {
       return state;
     }
+    // Some states return an array rather than an object. They need to be
```

Comment:
By "some states return an array" did you mean "some modules use an array as their state"?

Reply:
yes